### PR TITLE
Fix travis build

### PR DESCRIPTION
**Pull request by DorianGrey**
A "work in progress" for now - seems that not all changes made it into the merge with 1.1.1 because we to separate the stuff that should go into `react-scripts` 2.0 from what is required in 1.x.
Just testing this for the moment and see how far I may get ...

### CODE REVIEW

**Comment by DorianGrey** [comment]
@wmonk , things are getting green - yay!
A second look would be good, to take care I did not pick up something unintended. Stuff was picked from CRA's master branch that should not contain things planned for their 2.0 exclusively.

**Comment by DorianGrey** [comment]
Thanks for the review, I'm looking to fix these issue after my weekend vacation.

**Comment by DorianGrey** [comment]
I've disabled the `installs` test for now, since this will require a lot of additional work to be done.
I'm still working on the other ones -  esp. `e2e-simple.sh` was changed a lot, now using a local registry instead of local packing. Still experimenting which parts have to reverted to the previous strategy.

File: .travis.yml [hunk 1]
```diff
@@ -16,13 +12,17 @@ cache:
 install: true
 script:
  - 'if [ $TEST_SUITE = "simple" ]; then tasks/e2e-simple.sh; fi'
+ - 'if [ $TEST_SUITE = "installs" ]; then tasks/e2e-installs.sh; fi'
```
[the remark this answers is not in the document]
Comment (DorianGrey) [reply .travis.yml]: Done.
`install` is disabled for now.

File: .travis.yml [hunk 2]
```diff
@@ -16,13 +12,17 @@ cache:
...
-  global:
-    - USE_YARN=no
   matrix:
     - TEST_SUITE=simple
-    - TEST_SUITE=install
-# There's a weird Yarn/Lerna bug related to prerelease versions.
-# TODO: reenable after we ship 1.0.
-#    - node_js: 6
-#      env: USE_YARN=yes TEST_SUITE=simple
+    - TEST_SUITE=installs
```
[the remark this answers is not in the document]
Comment (DorianGrey) [reply .travis.yml]: Done.
`install` is disabled for now.

File: .travis.yml [hunk 3]
```diff
@@ -16,13 +12,17 @@ cache:
...
+    - TEST_SUITE=kitchensink
+matrix:
+  include:
+    - node_js: 0.10
```
[the remark this answers is not in the document]
Comment (DorianGrey) [reply .travis.yml]: Done.

File: tasks/e2e-installs.sh [hunk 4]
```diff
@@ -72,98 +72,82 @@ set -x
...
 # ******************************************************************************
-# First, pack and install create-react-app.
+# Test --scripts-version with a version number
 # ******************************************************************************
 
-# Pack CLI
-cd "$root_path"/packages/create-react-app
-cli_path=$PWD/`npm pack`
+cd "$temp_app_path"
+npx create-react-app --scripts-version=1.0.17 test-app-version-number
```
[the remark this answers is not in the document]
Comment (DorianGrey) [reply tasks/e2e-installs.sh]: This test suite is disabled for now.

File: tasks/e2e-simple.sh [hunk 5]
```diff
@@ -175,55 +133,25 @@ exists build/static/media/*.svg
...
-# the CLI properly in the temporary location if it is missing.
-npm init --yes
-
-# Now we can install the CLI from the local package.
-npm install "$cli_path"
-
 # Install the app in a temporary location
 cd $temp_app_path
-create_react_app --scripts-version="$scripts_path" test-app
+npx create-react-app test-app
```
[the remark this answers is not in the document]
Comment (DorianGrey) [reply tasks/e2e-simple.sh]: Done. Due to the new strategy with a local registry, it seems that we only need to provide `--scripts-version=react-scripts-ts` here.

File: tasks/e2e-simple.sh [hunk 6]
```diff
@@ -284,40 +212,39 @@ function verify_module_scope {
   echo "{}" >> sample.json
 
   # Save App.js, we're going to modify it
-  cp src/App.tsx src/App.tsx.bak
+  cp src/App.js src/App.js.bak
```
[the remark this answers is not in the document]
Comment (DorianGrey) [reply tasks/e2e-simple.sh]: Done.

**Comment by wmonk** [comment]
Awesome work @DorianGrey - is this ready to be merged?

**Comment by DorianGrey** [comment]
Suppose, yes.

We might have to deal with `e2e-installs.sh`, but I don't think it's that important for now. I've just disabled the test suite and left a particular comment that additional work has to be done before activating this again.

**Pull request merged**